### PR TITLE
Fix bugs for dropdown and date picker

### PR DESCRIPTION
**Pull request by fragsalat**
- Added className property to dropdown items
- Added title property to dropdown items
- Fixed behavior of selected items
- Fixed handling of property changes
- Added onChange property to propagate changes in react world
- Fixed change event and styling of date picker

### CODE REVIEW

File: src/ws-date-picker/ws-date-picker.js [hunk 1]
```diff
@@ -89,7 +92,7 @@ export class WSDatePicker extends Component {
    */
   onChange([selectedDate], value) {
     this.setState({value});
-    this.element.dispatchEvent(new CustomEvent('change', {detail: selectedDate}));
+    this.element.dispatchEvent(new CustomEvent('change', {detail: {date: selectedDate, value}, bubbles: true}));
     // Propagate if wanted
     if (this.props.onChange) {
       this.props.onChange(selectedDate);
```
Comment (fokusferit) [inline src/ws-date-picker/ws-date-picker.js]: It looks like the properties of ws-date-picker are not created, could you ? Or at least we should track the lack of documentation in an Issue

File: src/ws-date-picker/ws-date-picker.scss [hunk 2]
```diff
@@ -263,7 +264,7 @@
     font-size: rem(14px);
     width: rem(32px);
     height: rem(32px);
-    line-height: rem(32px);
+    line-height: rem(28px); // Numbers on acumin-pro font are lower
```
Comment (fokusferit) [inline src/ws-date-picker/ws-date-picker.scss]: A variable here or some calculation might be better I think

File: src/ws-dropdown/dropdown-menu-item.js [hunk 3]
```diff
@@ -89,9 +93,16 @@ export class DropdownMenuItem extends Component {
       this.props.handle('show-child', this.menu);
     } else {
       if (!this.context.multiple) {
-        this.state.selected = true;
-        this.props.handle('change', this.state);
+        // If it is selected we publish null because it will be deselected in the upper menu
+        if (this.state.selected) {
+          this.props.handle('change', null);
```
Comment (fokusferit) [inline src/ws-dropdown/dropdown-menu-item.js]: Two lines below you do `this.state.stored = true;` do we need to reset it here maybe ? idk what 'stored' exactly means, just asking to understand it properly :D

File: src/ws-dropdown/dropdown-menu.js [hunk 4]
```diff
@@ -318,8 +340,8 @@ export class DropdownMenu extends Component {
           />,
           <li className="dropdown-item-separator" key="parent-separator" />
         ]}
-        {(this.state.value && this.state.value.length) ? [
-          this.state.value.map((item, index) =>
+        {(this.state.value && this.state.value.length && (this.context.multiple || this.props.filterable)) ? [
```
Comment (fokusferit) [inline src/ws-dropdown/dropdown-menu.js]: This looks like complicated and hard to read. Maybe refactoring it into a function like "renderXyZ()" so it is easier to understand parts of the render function more easily ?

File: src/ws-dropdown/ws-dropdown.js [hunk 5]
```diff
@@ -37,7 +39,8 @@ export class WSDropdown extends Component {
     limit: 10,
     orientation: 'left',
     placeholder: '',
-    value: null
+    value: null,
+    onChange: () => {}
```
Comment (fokusferit) [inline src/ws-dropdown/ws-dropdown.js]: We need a jsDoc for that! :)

File: src/ws-date-picker/ws-date-picker.scss (hunk 2 above)
Comment (fragsalat) [inline src/ws-date-picker/ws-date-picker.scss]: If you tell me how to calculate the line height for this specific font I gonna use it ;) :D

File: src/ws-date-picker/ws-date-picker.js (hunk 1 above)
Comment (fragsalat) [inline src/ws-date-picker/ws-date-picker.js]: Actually I don't really got what you mean with the properties are not created.

File: src/ws-dropdown/dropdown-menu-item.js (hunk 3 above)
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu-item.js]: Nope. I'm pretty sure that we don't need to do it. The flag stored is an indicator that the item is selected to be presented as a value or stored in values list.

File: src/ws-dropdown/ws-dropdown.js (hunk 5 above)
Comment (fragsalat) [inline src/ws-dropdown/ws-dropdown.js]: Fixed

**Comment by tonysaad** [comment]
👍 I've tested it and it works well I will add something later for 'filterable' dropdowns that the input field gets auto-focused when the dropdown opens

**Pull request merged**